wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -1172,13 +1300,55 @@ int wc_iotsafe_ecc_verify_hash(byte *sig, word32 siglen, byte *hash,
     ret = wc_ecc_sig_to_rs(sig, siglen, sig_raw, &r_size, sig_raw
             + IOTSAFE_ECC_KSIZE, &s_size);
     if (ret == 0)
-        ret = iotsafe_verify_hash(key_id, IOTSAFE_HASH_SHA256, IOTSAFE_SIGN_ECDSA,
+        ret = iotsafe_verify_hash(key_id, id_size, IOTSAFE_HASH_SHA256, IOTSAFE_SIGN_ECDSA,
             hash, hashlen, sig_raw, 2 * IOTSAFE_ECC_KSIZE, res);
     return ret;
 }
+
+int wc_iotsafe_ecc_gen_k_ex(byte *key_id, uint16_t id_size)
+{
+    int ret = 0;
+    ecc_key* key = XMALLOC(sizeof(ecc_key), NULL, DYNAMIC_TYPE_ECC);
```

Comment:
This seems odd. Allocate from heap an `ecc_key`, not checking for null, not memset, not using the populated public ket `iotsafe_parse_public_key`... Perhaps a better solution is to allow calling `iotsafe_gen_keypair` with a NULL ecc_key?

Reply:
Excellent point. `iotsafe_gen_keypair` will now always return '0' when the `key` argument is NULL'ed. This will also save one iotsafe call to retrieve the public key if the caller does not need it.